feat: Add plugins slots for sidebars and application (#1752)
openedx/frontend-app-authoring: src/plugin-slots/CourseAuthoringOutlineSidebarSlot/README.md

# CourseAuthoringOutlineSidebarSlot

### Slot ID: `course_authoring_outline_sidebar_slot`

### Plugin Props:

* `courseId` - String. 

## Description

The slot wraps the sidebar that is displayed on the course outline page. It can
be used to add additional sidebar components or modify the existing sidebar.

## Example

![Screenshot of the outline sidebar surrounded by border](./images/outline_sidebar_with_border.png)

> The following example configuration inserts an extra button to the header as shown above.

Reviewer: ```suggestion
The following example configuration surrounds the sidebar in a border as shown above.
```
